src/backend/distributed/operations/citus_tools.c

```diff
@@ -11,25 +11,28 @@
  */
 
 #include "postgres.h"
-#include "funcapi.h"
-#include "libpq-fe.h"
-#include "miscadmin.h"
-
 
 #include "access/htup_details.h"
 #include "catalog/pg_type.h"
+#include "distributed/argutils.h"
 #include "distributed/connection_management.h"
 #include "distributed/metadata_cache.h"
+#include "distributed/multi_client_executor.h"
 #include "distributed/multi_server_executor.h"
 #include "distributed/remote_commands.h"
-#include "distributed/worker_protocol.h"
 #include "distributed/version_compat.h"
+#include "distributed/worker_protocol.h"
+#include "funcapi.h"
 #include "lib/stringinfo.h"
+#include "libpq-fe.h"
+#include "miscadmin.h"
 #include "utils/builtins.h"
 
-#include "distributed/multi_client_executor.h"
 
+/* simple query to run on workers to check connectivity */
+#define CONNECTIVITY_CHECK_QUERY "SELECT 1"
```

Comment:
minor: maybe HEALTH_CHECK_QUERY. 

I wonder would it ever make sense to set this query by the user. Say, instead of `SELECT 1`, a more specific checks implemented by a UDF `SELECT check_my_custom_health_check()` or such.

Anyway, we don't have to do that now, it might be useful to keep that in mind just in case we want to do that in the future.

Reply:
I think we can have a GUC that contains the name of the user defined health check function, and expect the user to create and distribute that function. If that GUC is not set, we can fallback to running `SELECT 1` as we do it now.

I will not work on this now. However it does not seem like too much work.

Reply:
> I will not work on this now. However it does not seem like too much work.

I guess verifying the return type etc. might be complicated, I think we can skip it for now?

Reply:
I thought about this, and I think it is best to keep this idea of user defined health check functions out of scope.